utils/protocols.py

```diff
@@ -1,13 +1,32 @@
-ONTOLOGY_10x = ["EFO:0009310", "EFO:0009899"]
+from api.ontology import OntologyAPI
 
+ONTOLOGY_10x_PARENT = "EFO:0008995"
+ONTOLOGY_10xV2_PARENT = "EFO:0009310"
+ONTOLOGY_10xV3_PARENT = "EFO:0009898"
 
-def is_10x(library_preparation_protocol):
+
+def is_10x(ontology_api: OntologyAPI, library_preparation_protocol : dict):
+    library_construction = get_library_construction(library_preparation_protocol)
+    return ontology_api.is_equal_or_descendant(ONTOLOGY_10x_PARENT, library_construction)
+
+
+def map_10x_bam_schema(ontology_api: OntologyAPI, library_preparation_protocol: dict):
+    library_construction = get_library_construction(library_preparation_protocol)
+    if ontology_api.is_equal_or_descendant(ONTOLOGY_10xV2_PARENT, library_construction):
+        return '10xV2'
+    if ontology_api.is_equal_or_descendant(ONTOLOGY_10xV3_PARENT, library_construction):
+        return '10xV3'
+    return None
+
+
+def get_library_construction(library_preparation_protocol: dict):
     content = library_preparation_protocol.get("content", {})
-    library_const_approach_obj = content.get("library_construction_approach",
-                                             content.get("library_construction_method", {}))
-    library_const_approach = library_const_approach_obj.get('ontology')
+    library_construction = content.get("library_construction_approach", content.get("library_construction_method", {}))
+    library_construction_ontology = library_construction.get('ontology')
+    if not library_construction_ontology:
+        raise ProtocolError(f'Could not determine library construction from library_preparation_protocol: {library_preparation_protocol}')
```

Comment:
Sorry I'm being nitpicky here, i think here it's fine to return null here, and have a check if it exists in the archiver.py and throw the error there. The param to is10x will be the library construction approach. Please feel free to not follow this suggestion for the sake of time.

Reply:
Thanks for making the changes!

Reply:
I take back what i said. This is totally fine, since input is the schema. Please disregard my comments.